receiver/prometheusreceiver/metrics_receiver_test_helper.go

```diff
@@ -0,0 +1,452 @@
+		default:
+		}
+	}
+	return n
+}
+
+type metricTypeComparator func(*testing.T, *pdata.Metric) bool
+type numberPointComparator func(*testing.T, *pdata.NumberDataPoint) bool
+type histogramPointComparator func(*testing.T, *pdata.HistogramDataPoint) bool
+type summaryPointComparator func(*testing.T, *pdata.SummaryDataPoint) bool
+
+type dataPointExpectation struct {
+	numberPointComparator    []numberPointComparator
+	histogramPointComparator []histogramPointComparator
+	summaryPointComparator   []summaryPointComparator
+}
+
+type testExpectation func(*testing.T, *pdata.ResourceMetrics) bool
+
+func doCompare(name string, t *testing.T, want pdata.AttributeMap, got *pdata.ResourceMetrics, expectations []testExpectation) {
+	t.Run(name, func(t *testing.T) {
+		assert.Equal(t, expectedScrapeMetricCount, countScrapeMetricsRM(got))
+		assert.Equal(t, want.Len(), got.Resource().Attributes().Len())
+		for k, v := range want.AsRaw() {
+			value, _ := got.Resource().Attributes().Get(k)
+			assert.EqualValues(t, v, value.AsString())
+		}
+		for _, e := range expectations {
+			assert.True(t, e(t, got))
+		}
+	})
+}
+
+func assertMetricPresent(name string, metricTypeExpectations metricTypeComparator,
+	dataPointExpectations []dataPointExpectation) testExpectation {
+	return func(t *testing.T, rm *pdata.ResourceMetrics) bool {
+		allMetrics := getMetrics(rm)
+		for _, m := range allMetrics {
+			if name != m.Name() {
+				continue
+			}
+			if !metricTypeExpectations(t, m) {
+				return false
+			}
+			for i, de := range dataPointExpectations {
+
+				for _, npc := range de.numberPointComparator {
+					switch m.DataType() {
+					case pdata.MetricDataTypeGauge:
+						dataPoint := m.Gauge().DataPoints().At(i)
+						if !npc(t, &dataPoint) {
+							return false
+						}
+					case pdata.MetricDataTypeSum:
+						dataPoint := m.Sum().DataPoints().At(i)
+						if !npc(t, &dataPoint) {
+							return false
+						}
+					}
+				}
+
+				switch m.DataType() {
+				case pdata.MetricDataTypeHistogram:
+					for _, hpc := range de.histogramPointComparator {
+						dataPoint := m.Histogram().DataPoints().At(i)
+						if !hpc(t, &dataPoint) {
+							return false
+						}
+					}
+				case pdata.MetricDataTypeSummary:
+					for _, spc := range de.summaryPointComparator {
+						dataPoint := m.Summary().DataPoints().At(i)
+						if !spc(t, &dataPoint) {
+							return false
+						}
+					}
+				}
+			}
+			return true
+		}
+		assert.Failf(t, "Unable to match metric expectation", name)
+		return false
+	}
+}
+
+func assertMetricAbsent(name string) testExpectation {
+	return func(t *testing.T, rm *pdata.ResourceMetrics) bool {
+		allMetrics := getMetrics(rm)
+		for _, m := range allMetrics {
+			if !assert.NotEqual(t, name, m.Name()) {
+				return false
+			}
+		}
+		return true
+	}
+}
+
+func compareMetricType(typ pdata.MetricDataType) metricTypeComparator {
+	return func(t *testing.T, metric *pdata.Metric) bool {
+		return assert.Equal(t, typ, metric.DataType())
```

Comment:
Please add some messages to these assertions to help identify where test failures come from.

Reply:
Added messages in all assert.Equal of comparators methods.